raphtory/src/lib.rs

```diff
@@ -187,18 +192,339 @@ mod test_utils {
         )
     }
 
+    pub(crate) fn prop(p_type: &PropType) -> impl Strategy<Value = Prop> {
+        match p_type {
+            PropType::Str => any::<String>().prop_map(|s| Prop::str(s)).boxed(),
+            PropType::I64 => any::<i64>().prop_map(|i| Prop::I64(i)).boxed(),
+            PropType::F64 => any::<f64>().prop_map(Prop::F64).boxed(),
+            PropType::U8 => any::<u8>().prop_map(Prop::U8).boxed(),
+            PropType::Bool => any::<bool>().prop_map(Prop::Bool).boxed(),
+            PropType::DTime => (1970..2024, 1..=12, 1..28, 0..24, 0..60, 0..60)
```

Comment:
might might be a good idea to tests dates before 1970 as they end up with negative timestamps?